D110-D113 MSTCNV instrumentation (call-boundary dump ffc_mc_cols, stage checkpoints ffc_mc_ck; hooks ffm_const/ffm_in/ffm_ck/ffm_out). Generated by diff -u against the PRISTINE CLOUDS2.F90 (2026-10-05). Hunks are local: a declaration block after 'integer :: lborrow1' in MSTCNV, the entry hook before 'ierr=0 ; lerr=0', nine checkpoint hooks and the exit hook before the final 'return'. Field layouts are in fullfidelity/clouds_mstcnv_io.py (the array constructors were generated from its tables). Apply with ATM_DRV_clouds_mstcnv.f.patch (helpers; units 1200-1202).

--- CLOUDS2.F90.orig	2026-10-05 16:19:35.089065970 -0400
+++ CLOUDS2.F90	2026-10-05 16:42:56.901011871 -0400
@@ -729,6 +729,8 @@
 
     real*8 :: vsum
     integer :: lborrow1
+    logical :: ffm_ckon   ! full-fidelity-port instrumentation (D110)
+    integer :: ffm_ks(4)
 
 #ifdef TRACERS_ON
 !@var TMOLD: old TM (tracer mass)
@@ -821,6 +823,15 @@
     !**** (7) CONVECTIVE CLOUD OPTICAL THICKNESS LOOP, THE ONLY ONE OUTSIDE
     !****     THE MAJOR OUTER LOOP.
     !****
+    ffm_ks=(/1,1+(KMAX-1)/3,1+2*(KMAX-1)/3,KMAX/)
+    call ffm_const(26, (/ entrainment_cont1, entrainment_cont2, radiusl_multiplier, radiusi_multiplier, U00a, U00b,  &
+         wmu_multiplier, RWCldOX, RIMAX, MC_FDDRT, dble(MC_ENTR_MASS_LIM_PLUME),  &
+         dble(MC_NEW_DDRFT_THETAV), dble(MC_REVP_ABV_CLDBASE), CCMUL, CCMUL2, COETAU, WMU, WMUL,  &
+         TI, CLDMIN, FDDET, DTMIN1, BRCLD, dble(LMCM), dble(LM), SECONDS_PER_HOUR /))
+    call ffm_in(i_debug,j_debug,ffm_ckon,1974, &
+         (/ PEARTH, PLAND, dble(DCL), dble(LMCM), XMASS, BYDTsrc, DTsrc, BYBR, dble(KMAX), PL, PLE,  &
+         PLK, AIRM, BYAM, ETAL, TL, TVL, SM, QM, QCLL, QCIL, SDL, WTURB, GZL, SMOM, QMOM,  &
+         RA(ffm_ks), UM(ffm_ks,:), VM(ffm_ks,:), U_0(ffm_ks,:), V_0(ffm_ks,:) /))
     ierr=0 ; lerr=0
     LMCMIN=0
     LMCMAX=0
@@ -1049,6 +1060,8 @@
       SLH=LHX*BYSHA
       DMSE=(SVUP-SVEDG)*PLK(LMIN+1)+(SVEDG-SVDN)*PLK(LMIN)+ &
            SLH*(QSAT(SUP*PLK(LMIN+1),LHX,PL(LMIN+1))-QDN)
+      if (ffm_ckon) call ffm_ck(1,LMIN,0,0,2, &
+           (/ DMSE, FMP0 /))
       if(DMSE.gt.-1d-10) cycle  ! try next level
 
       !**** MASS_FLUX PERFORMS THE ITERATIONS
@@ -1056,6 +1069,8 @@
       call MASS_FLUX (FPLUME, FMP2, DQSUM, LMIN, &
            LHX, QMO1, QMO2, SLH, SMO1, SMO2, WMDN, WMUP, WMEDG)
 
+      if (ffm_ckon) call ffm_ck(2,LMIN,0,0,3, &
+           (/ FPLUME, FMP2, DQSUM /))
       if(FPLUME.le..001) cycle ! try next level
 
       !****
@@ -1177,6 +1192,8 @@
 
           !**** adjust MPLUME to take account of restricted area of subsidence
           !**** (i.e. MPLUME is now a greater fraction of the relevant airmass.
+          if (ffm_ckon) call ffm_ck(3,LMIN,IC,NPPL,6, &
+               (/ FCTYPE, MPLUM1, MPLUME, FMC1, dble(MCCONT), merge(1d0,0d0,MC1) /))
           MPLUME = Min (MPLUME/FMC1, AIRM(LMIN)*0.95d0*QM(LMIN)/(QMOLD(LMIN) + teeny))
           if(MPLUME.le..001*AIRM(LMIN)) cycle CLOUD_TYPES
           FPLUME=MPLUME*BYAM(LMIN)
@@ -1969,6 +1986,14 @@
 #endif
 
           end do  CLOUD_TOP     !   End Loop 3  (L)
+          if (ffm_ckon) call ffm_ck(4,LMIN,IC,NPPL,3821, &
+               (/ dble(LMAX), dble(MCCONT), merge(1d0,0d0,MC1), FMC1, dble(LDRAFT), dble(LFRZ), ETADN,  &
+               MPLUME, FPLUME, SMP, QMP, merge(MPMAX,0d0,LMAX.gt.LMIN), merge(SMPMAX,0d0,LMAX.gt.LMIN),  &
+               merge(QMPMAX,0d0,LMAX.gt.LMIN), CDHSUM, CDHSUM1, CDHDRT, DM, DMR, DDR, CCM, COND, CONDP,  &
+               CONDP1, CONDV, TAUMC1, CDHEAT, ENT, DET, BUOY, WCU, TPSAV, SMDNL, QMDNL, DSM, DSMR, DQM,  &
+               DQMR, VLAT, U00L, CONDMMR, DSMOM, DSMOMR, DQMOM, DQMOMR, SMOMDNL, QMOMDNL, SMOMP, QMOMP,  &
+               merge(SMOMPMAX,0d0*SMOMP,LMAX.gt.LMIN), merge(QMOMPMAX,0d0*QMOMP,LMAX.gt.LMIN),  &
+               UMP(ffm_ks), VMP(ffm_ks), DUM(ffm_ks,:), DVM(ffm_ks,:), UMDNL(ffm_ks,:), VMDNL(ffm_ks,:) /))
 
           if(LMIN.eq.LMAX) then
 
@@ -2264,6 +2289,12 @@
           end do
           DM(LDMIN)=DM(LDMIN)+DDRAFT
         end if
+        if (ffm_ckon) call ffm_ck(5,LMIN,IC,0,2249, &
+             (/ dble(LDMIN), dble(LLMIN), merge(EDRAFT,0d0,ETADN.gt.1d-10),  &
+             merge(DDRAFT,0d0,ETADN.gt.1d-10), merge(SMDN,0d0,ETADN.gt.1d-10),  &
+             merge(QMDN,0d0,ETADN.gt.1d-10), EVPSUM, CDHSUM1, ETADN, DM, DSM, DQM, DMR, DSMR, DQMR,  &
+             DDM, COND, CDHEAT, TAUMCL, TDNL, QDNL, DSMOM, DQMOM, DSMOMR, DQMOMR, DUM(ffm_ks,:),  &
+             DVM(ffm_ks,:) /))
 
         !****
         !**** SUBSIDENCE AND MIXING LOOP (5)
@@ -2453,6 +2484,9 @@
 #endif  /* TRACERS_ON */
 
         end do        ! end of sub-timesteps for subsidence
+        if (ffm_ckon) call ffm_ck(6,LMIN,IC,0,1206, &
+             (/ dble(KSUB), dble(IERR), dble(LERR), dble(LDMIN), dble(LMAX), CM, CMNEG, SM, QM, SMOM,  &
+             QMOM, UM(ffm_ks,:), VM(ffm_ks,:) /))
 
 
 !#ifndef WEAKER_MC_LIMITS
@@ -2669,6 +2703,9 @@
         If (PLE(LMIN)-PLE(LMAX+1) >= 450)  DPHADEEP(LMAX) = DPHADEEP(LMAX) + (CDHSUM-CDHSUM1+CDHM)*FMC1
 
 !**** Loop down from top of plume
+        if (ffm_ckon) call ffm_ck(7,LMIN,IC,0,884, &
+             (/ PRCP, PRHEAT, TOLD, COND, CONDP, HEAT1, SVLATL, SVWMXL, CONDPT, MCFLX, DGDSM, DQMTOTAL,  &
+             SM, QM, VLAT, LHP, SMOM /))
 EVAP_PRECIP: do L=LMAX-1,1,-1
 
           !**** CALCULATE CONVECTIVE CLOUD FRACTION
@@ -2930,6 +2967,10 @@
         !****
         !**** END OF INNER LOOP (2) OVER CLOUD TYPES
         !****
+        if (ffm_ckon) call ffm_ck(8,LMIN,IC,0,1370, &
+             (/ PRCP, PRCPMC, CLDSLWIJ, CLDDEPIJ, dble(LMCMIN), dble(LMCMAX), PRHEAT, FMC1, CLDMCL,  &
+             PRECNVL, LHP, HEAT1, COND, SM, QM, DPHASE, DQCOND, DQCTOTAL, DGDSM, DDMFLX, SVLATL,  &
+             SVWMXL, CONDPT, DQMTOTAL, SMOM, QMOM /))
         MC1=.false.
 
 #ifdef TRACERS_ON
@@ -2989,6 +3030,8 @@
 
     end if
 
+    if (ffm_ckon) call ffm_ck(9,0,0,0,124, &
+         (/ dble(LMCMIN), dble(LMCMAX), FMC1, merge(AIRXL,0d0,LMCMIN.gt.0), FSSL, DGDSM, SM /))
     !****
     !**** CALCULATE CONVECTIVE CLOUD OPTICAL THICKNESS
     !****
@@ -3173,6 +3216,13 @@
     END DO
 #endif
 
+    call ffm_out(2616, &
+         (/ dble(IERR), dble(LERR), dble(LMCMIN), dble(LMCMAX), PRCPMC, CLDSLWIJ, CLDDEPIJ, AIRXL,  &
+         PRHEAT, WMSUM, WMCTWP, WMCLWP, FMC1, dble(MCCONT), TL, SM, QM, FSSL, CLDMCL, TAUMCL,  &
+         SVLATL, SVLAT1, SVWMXL, CSIZEL, CONDPT, VSUBL, TPSAV, MCFLX, DGDSM, DGDEEP, DGSHLW,  &
+         DPHASE, DPHADEEP, DPHASHLW, DTOTW, DQCOND, DGDQM, DQMTOTAL, DQMSHLW, DQMDEEP, DQCTOTAL,  &
+         DQCSHLW, DQCDEEP, DDMFLX, TDNL, QDNL, U00L, QLmc, QImc, CNVMMRL, CONDMMR, LHP, PRECNVL,  &
+         SMOM, QMOM, UM(ffm_ks,:), VM(ffm_ks,:) /))
     return
 
 #ifdef TRACERS_ON
